### PR TITLE
feat(services): migrate preview env react query 

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-920

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Programs must be written for people to read, and only incidentally for machines to execute

File: libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx [hunk 1]
```diff
@@ -40,28 +30,37 @@ export function PageSettingsPreviewEnvironmentsFeature() {
...
-          await dispatch(
-            editApplication({
-              applicationId: application.id,
-              data: cloneApplication,
-              serviceType: getServiceType(application),
-              silentToaster: true,
-              // eslint-disable-next-line @typescript-eslint/no-empty-function
-              toasterCallback: () => {},
-              queryClient,
+      services?.forEach(async (service: AnyService) => {
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx:39]:
```suggestion
      services.forEach(async (service: AnyService) => {
```

File: libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx [hunk 2]
```diff
@@ -40,28 +30,37 @@ export function PageSettingsPreviewEnvironmentsFeature() {
...
+          }
+
+          const payload = match(service)
+            .with({ serviceType: 'APPLICATION' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'CONTAINER' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'JOB' }, (s) => buildEditServicePayload({ service: s, request }))
+            .with({ serviceType: 'HELM' }, (s) => buildEditServicePayload({ service: s, request }))
+            .otherwise(() => null)
+
+          if (!payload) return
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx:54]:
```suggestion
            .exhaustive()
```
as you have handle `DATABASE` type upfront it should be fine

File: libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx [hunk 3]
```diff
@@ -73,13 +72,15 @@ export function PageSettingsPreviewEnvironmentsFeature() {
     methods.setValue('on_demand_preview', value)
     //set all preview applications "true" when env preview is true
     if (loadingStatusEnvironmentDeploymentRules) {
-      applications?.forEach((application) => methods.setValue(application.id, value, { shouldDirty: true }))
+      services?.forEach((service) => methods.setValue(service.id, value, { shouldDirty: true }))
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx:75]:
```suggestion
      services.forEach((service) => methods.setValue(service.id, value, { shouldDirty: true }))
```

File: libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx [hunk 4]
```diff
@@ -91,18 +92,22 @@ export function PageSettingsPreviewEnvironmentsFeature() {
...
     // !loading is here to prevent the toggle to glitch the time we are submitting the two api endpoints
     if (environmentDeploymentRules && loadingStatusEnvironmentDeploymentRules && !loading) {
-      const isApplicationPreviewEnabled = applications ? applications.some((app) => app.auto_preview) : false
+      const isApplicationPreviewEnabled = services
+        ? services.some((app) => app.serviceType !== 'DATABASE' && app.auto_preview)
+        : false
       methods.setValue('auto_preview', environmentDeploymentRules.auto_preview || isApplicationPreviewEnabled)
       methods.setValue('on_demand_preview', environmentDeploymentRules.on_demand_preview)
-      applications?.forEach((application) => methods.setValue(application.id, application.auto_preview))
+      services?.forEach((service) =>
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx:100]:
```suggestion
      services.forEach((service) =>
```

File: libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
-  Icon,
-  InputToggle,
-} from '@qovery/shared/ui'
+import { BlockContent, Button, HelpSection, Icon, InputToggle } from '@qovery/shared/ui'
 
 export interface PageSettingsPreviewEnvironmentsProps {
   onSubmit: () => void
   loading: boolean
-  applications?: ApplicationEntity[]
+  services?: AnyService[]
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx:10]:
```suggestion
  services: AnyService[]
```

File: libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx [hunk 6]
```diff
@@ -68,49 +68,46 @@ export function PageSettingsPreviewEnvironments(props: PageSettingsPreviewEnviro
...
-                          toggleEnablePreview(value)
-                          field.onChange(value)
-                        }}
-                        title={
-                          <span className="flex items-center -top-1 relative">
-                            <Icon name={IconEnum.APPLICATION} className="mr-3" /> {application.name}
-                          </span>
-                        }
-                        small
+              {services?.map(
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx:77]:
```suggestion
              {services.map(
```

File: libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx [hunk 7]
```diff
@@ -68,49 +68,46 @@ export function PageSettingsPreviewEnvironments(props: PageSettingsPreviewEnviro
                 />
               )}
             />
-            <div data-testid="toggles" className={applications && applications.length > 0 ? 'mt-5' : ''}>
-              {applications && applications.length > 0 && (
-                <h2 data-testid="applications-title" className="font-medium text-neutral-400 text-ssm mb-5">
+            <div data-testid="toggles" className={services && services.length > 0 ? 'mt-5' : ''}>
+              {services && services.length > 0 && (
```
Comment (ctjhoa) [inline libs/pages/services/src/lib/ui/page-settings-preview-environments/page-settings-preview-environments.tsx:72]:
```suggestion
              {services.length > 0 && (
```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**